tests/unit/Common/Internal/Authentication/SharedAccessSignatureHelperTest.php

```diff
@@ -233,4 +233,377 @@ public function testGenerateAccountSharedAccessSignatureToken()
+        return $this->signedIP;
+    }
+    
+    public function getExpectedSignature() {
+        return $this->expectedSignature;
+    }
+
+    public static function BuildTestCases() {
+        $testCases = array();
+
+        // ?sv=2016-05-31&ss=bfqt&srt=sco&sp=rwdlacup&se=2017-03-24T21:14:01Z&st=2017-03-17T13:14:01Z&spr=https&sig=ZpEYbkT%2B9NJTYyMIuFnXQ9RzOehYF1mjnsk00B%2FX1nw%3D
+        $testCases[] = new GenerateAccountSASTestCase(
+            "2016-05-31", // signedVersion
+            "bfqt", // signedService
+            "sco", // signedResourceType
+            "rwdlacup", // signedPermission
+            "2017-03-24T21:14:01Z", // signedExpiracy
+            "2017-03-17T13:14:01Z", // signedStart
+            "https", // signedProtocol
+            "", // signedIP
+            "ZpEYbkT%2B9NJTYyMIuFnXQ9RzOehYF1mjnsk00B%2FX1nw%3D" // expectedSignature
+        );
+
+        // ?sv=2016-05-31&ss=bfqt&srt=sco&sp=rwdlacup&se=2017-03-24T21:14:01Z&st=2017-03-17T13:14:01Z&sip=168.1.5.65&spr=https,http&sig=GZcWRjLJk%2FJSbM9zKb1XufTt2OueTSSgwsa03nYn5yM%3D
+        $testCases[] = new GenerateAccountSASTestCase(
+            "2016-05-31", // signedVersion
+            "bfqt", // signedService
+            "sco", // signedResourceType
+            "rwdlacup", // signedPermission
+            "2017-03-24T21:14:01Z", // signedExpiracy
+            "2017-03-17T13:14:01Z", // signedStart
+            "https,http", // signedProtocol
+            "168.1.5.65", // signedIP
+            "GZcWRjLJk%2FJSbM9zKb1XufTt2OueTSSgwsa03nYn5yM%3D" // expectedSignature
+        );
+
+        // ?sv=2016-05-31&ss=bf&srt=s&sp=rw&se=2017-03-24T00:00:00Z&st=2017-03-17T00:00:00Z&spr=https&sig=1%2BAozefG5VZDx9XorEGrAjOiTS8dX%2BJelK5SW91Zvq0%3D
+        $testCases[] = new GenerateAccountSASTestCase(
+            "2016-05-31", // signedVersion
+            "bf", // signedService
+            "s", // signedResourceType
+            "rw", // signedPermission
+            "2017-03-24T00:00:00Z", // signedExpiracy
+            "2017-03-17T00:00:00Z", // signedStart
+            "https", // signedProtocol
+            "", // signedIP
+            "1%2BAozefG5VZDx9XorEGrAjOiTS8dX%2BJelK5SW91Zvq0%3D" // expectedSignature
+        );
+
+        // ?sv=2016-05-31&ss=q&srt=o&sp=up&se=2017-03-24T00:00:00Z&st=2017-03-17T00:00:00Z&spr=https&sig=k1BKI65TdXs7rdAJiqDSJ6wYHjfJD0CJgplvOyqBK7Y%3D
+        $testCases[] = new GenerateAccountSASTestCase(
+            "2016-05-31", // signedVersion
+            "q", // signedService
+            "o", // signedResourceType
+            "up", // signedPermission
+            "2017-03-24T00:00:00Z", // signedExpiracy
+            "2017-03-17T00:00:00Z", // signedStart
+            "https", // signedProtocol
+            "", // signedIP
+            "k1BKI65TdXs7rdAJiqDSJ6wYHjfJD0CJgplvOyqBK7Y%3D" // expectedSignature
+        );
+
+        return $testCases;
+    }
+}
+
+class GenerateServiceSASTestCase extends GenerateAccountSASTestCase
```

Comment:
Advice not to have two class in one file, and also, it does not make much sense for this class to inherit from `GenerateAccountSASTestCase`. Also please reference the updated test case for account SAS [here](https://github.com/Azure/azure-storage-php/blob/master/tests/unit/Common/Internal/Authentication/SharedAccessSignatureHelperTest.php#L207) for a better practise.